lib/python/rose/config_editor/main.py

```diff
@@ -1847,17 +1848,52 @@ def spawn_window(config_directory_path=None, debug_mode=False,
     splash_screen = rose.gtk.splash.SplashScreenProcess(logo, title,
                                                         number_of_events)
     try:
-        MainController(config_directory_path,
-                       load_updater=splash_screen,
-                       load_all_apps=load_all_apps,
-                       load_no_apps=load_no_apps,
-                       metadata_off=metadata_off)
+        ctrl = MainController(config_directory_path,
+                              load_updater=splash_screen,
+                              load_all_apps=load_all_apps,
+                              load_no_apps=load_no_apps,
+                              metadata_off=metadata_off)
+
     except BaseException as e:
         splash_screen.stop()
         if debug_mode and isinstance(e, Exception):
             # Write out origin information - this is otherwise lost.
             traceback.print_exc()
         raise e
+
+    # open up any initial_namespaces the user has provided us with
+    if initial_namespaces:
+        # if the namespace ends with a / remove it
+        for i in range(len(initial_namespaces)):
+            if (len(initial_namespaces[i]) > 1 and
+                    initial_namespaces[i][-1] == '/'):
+                initial_namespaces[i] = initial_namespaces[i][0:-1]
+
+        # for each partial namespace get the full namespace
+        full_namespaces = []
+        for namespace in initial_namespaces:
+            exp = re.compile('(.*%s?[^\/]+)' % (re.escape(namespace),))
+            for ns in ctrl.data.namespace_meta_lookup:
+                match = exp.search(ns)
+                if match:
+                    full_namespaces.append(match.groups()[0])
+                    break
+
+        # open each namespace in a new tab
+        for namespace in full_namespaces:
+            # if the namespace begins with a / remove it
+            if namespace[0] == '/':
+                namespace = namespace[1:]
+            # open namespace
+            try:
+                ctrl.view_page(namespace)
+            except:
```

Comment:
`except Exception` is better here - a bare except can catch KeyboardInterrupt and SystemExit.